clippy_lints/src/consts.rs

```diff
@@ -180,7 +180,7 @@ pub fn lit_to_constant(lit: &LitKind) -> Constant {
         LitKind::Byte(b) => Constant::Int(ConstInt::U8(b)),
         LitKind::ByteStr(ref s) => Constant::Binary(s.clone()),
         LitKind::Char(c) => Constant::Char(c),
-        LitKind::Int(value, LitIntType::Unsuffixed) => Constant::Int(ConstInt::Infer(value)),
+        LitKind::Int(value, LitIntType::Unsuffixed) => Constant::Int(ConstInt::U128(value as u128)),
```

Comment:
same here, this is wrong, it should use the same logic (for integers) as https://github.com/rust-lang/rust/blob/master/src/librustc_const_eval/eval.rs#L845 (and thus also take the same arguments)